### PR TITLE
BC-5414 - Add ui-card component

### PR DESCRIPTION
**Pull request by bischofmax**
# Short Description

## Checklist before merging

- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [ ] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by bergatco — DISMISSED** [review]

File: src/components/ui-card/Card.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+			{{ subtitle }}
+		</v-card-subtitle>
+	</div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent } from "vue";
+
+export default defineComponent({
+	name: "Card",
```
Comment (bischofmax) [inline src/components/ui-card/Card.vue:37]:
```suggestion
	name: "CardTitle",
```

Comment (bischofmax) [reply src/components/ui-card/Card.vue:37]: Any better suggestions?

File: src/components/feature-board-file-element/content/display/file-description/FileDescription.vue [hunk 2]
```diff
@@ -1,56 +1,41 @@
...
-					>{{ name }}</span
-				>
-			</div>
-		</v-card-title>
-
-		<v-card-subtitle v-if="caption && !isEditMode">
-			{{ caption }}
-		</v-card-subtitle>
-	</div>
+	<Card
```
Comment (SevenWaysDP) [inline src/components/feature-board-file-element/content/display/file-description/FileDescription.vue:2]: Other name

File: src/components/ui-card/Card.vue (hunk 1 above)
Comment (SevenWaysDP) [reply src/components/ui-card/Card.vue:37]: Yes is better or maby ContentElementTitle

File: src/components/ui-card/Card.vue [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<template>
+	<div class="grey lighten-4">
+		<v-card-title
+			v-if="title || icon"
+			data-testid="board-file-element-display"
```
Comment (SevenWaysDP) [inline src/components/ui-card/Card.vue:5]: other test id name

File: src/components/ui-card/Card.vue [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+			inactive
+		>
+			<div class="d-flex flex-no-wrap align-center text-truncate">
+				<v-icon
+					v-if="icon"
+					color="black"
+					class="mr-2"
+					size="18"
+					data-testid="board-file-element-display-file-icon"
```
Comment (SevenWaysDP) [inline src/components/ui-card/Card.vue:14]: other test id name

File: src/components/ui-card/Card.vue [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+					>{{ icon }}</v-icon
+				>
+
+				<span
+					v-if="title"
+					:class="spanClasses"
+					data-testid="board-file-element-display-file-name"
```
Comment (SevenWaysDP) [inline src/components/ui-card/Card.vue:21]: other test id name

File: src/components/ui-card/Card.vue [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	props: {
+		title: { type: String, required: false },
+		subtitle: { type: String, required: false },
+		icon: { type: String, required: false },
+		shortenTitle: { type: Boolean, required: false },
+	},
+	setup(props) {
+		const spanClasses = computed(() => {
+			const marginClass = props.shortenTitle ? " mr-10" : "";
+			const classes = `subtitle-1 font-weight-bold text-truncate${marginClass}`;
```
Comment (SevenWaysDP) [inline src/components/ui-card/Card.vue:47]: maby it is not always mr-10 ?

File: src/components/ui-card/Card.vue (hunk 1 above)
Comment (bischofmax) [reply src/components/ui-card/Card.vue:37]: Frontend Arc decided not to have generic components for specific contexts. So this wouldn't be allowed in the folder structure.

File: src/components/ui-card/Card.vue (hunk 6 above)
Comment (bischofmax) [reply src/components/ui-card/Card.vue]: Should be removed in 3DotMenu Refactoring

**Review by SevenWaysDP — APPROVED** [review]

File: src/components/ui-card-title/CardTitle.vue [hunk 7]
```diff
@@ -0,0 +1,53 @@
+<template>
+	<div class="grey lighten-4">
```
Comment (bischofmax) [inline src/components/ui-card-title/CardTitle.vue:2]: Add white background in EditMode

File: src/components/ui-card-title/CardTitle.vue [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+				<span
+					v-if="title"
+					:class="spanClasses"
+					data-testid="card-title-string"
+					>{{ title }}</span
+				>
+			</div>
+		</v-card-title>
+
```
Comment (bischofmax) [inline src/components/ui-card-title/CardTitle.vue:22]: Add slot for menu

**Pull request closed without merging**